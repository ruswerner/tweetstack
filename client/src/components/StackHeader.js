import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {Toolbar} from 'material-ui/Toolbar';
import TextField from 'material-ui/TextField';
import IconButton from 'material-ui/IconButton';
import NavigationCloseIcon from 'material-ui/svg-icons/navigation/close';
import NavigationRefreshIcon from 'material-ui/svg-icons/navigation/refresh';

export default class StackHeader extends PureComponent {

  static propTypes = {
    id: PropTypes.string.isRequired,
    query: PropTypes.string.isRequired,
    canRefresh: PropTypes.bool.isRequired,
    loading: PropTypes.bool.isRequired,
    onSearch: PropTypes.func.isRequired,
    onRefresh: PropTypes.func,
    onRemove: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {
      query: props.query,
      focus: false
    };
    this.input = null;
  }

  handleChange = event => this.setState({query: event.target.value});
  handleFocus = () => this.setState({focus: true});
  handleBlur = () => {
    const newQuery = this.state.query.trim();
    if (newQuery.length === 0) {
      this.setState({focus: false, query: this.props.query});
      return;
    }
    if (newQuery !== this.props.query) {
      this.props.onSearch(newQuery);
    }
    this.setState({focus: false});
  };
  handleKey = event => {
    if (event.key === 'Enter' && this.input) {
      this.input.blur();
    }
  };

  componentWillReceiveProps(nextProps) {
    // Reset the state's query if the incoming prop changes
    if (this.props.query !== nextProps.query) {
      this.setState({query: nextProps.query});
    }
  }

  componentDidMount() {
    if (this.input) {
      this.input.focus();
    }
  }

  componentWillUnmount() {
    this.input = null;
  }

  render() {
    return (
      <Toolbar style={{paddingRight: 0}}>
        <TextField
          id={this.props.id}
          hintText="Search..."
          value={this.state.query}
          disabled={this.props.loading}
          onChange={this.handleChange}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
          onKeyDown={this.handleKey}
          ref={n => this.input = n}
        />
        <div>
          {!this.state.focus && this.props.canRefresh &&
           <IconButton onClick={this.props.onRefresh}>
             <NavigationRefreshIcon/>
           </IconButton>
          }
          <IconButton onClick={this.props.onRemove}>
            <NavigationCloseIcon/>
          </IconButton>
        </div>
      </Toolbar>
    );
  }

}
