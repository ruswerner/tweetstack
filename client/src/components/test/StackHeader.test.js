import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import StackHeader from '../StackHeader';
import renderer from 'react-test-renderer';

const noop=()=>null;

describe('StackHeader', function() {

  it('renders', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <StackHeader
          id="1"
          query="q1"
          canRefresh={false}
          loading={false}
          onSearch={noop}
          onRefresh={noop}
          onRemove={noop}
        />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders canRefresh', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <StackHeader
          id="1"
          query="q1"
          canRefresh={true}
          loading={false}
          onSearch={noop}
          onRefresh={noop}
          onRemove={noop}
        />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders loading', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <StackHeader
          id="1"
          query="q1"
          canRefresh={false}
          loading={true}
          onSearch={noop}
          onRefresh={noop}
          onRemove={noop}
        />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders canRefresh & loading', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <StackHeader
          id="1"
          query="q1"
          canRefresh={true}
          loading={true}
          onSearch={noop}
          onRefresh={noop}
          onRemove={noop}
        />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
