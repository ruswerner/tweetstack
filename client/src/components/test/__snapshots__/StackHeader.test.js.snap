// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackHeader renders 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "rgb(232, 232, 232)",
      "boxSizing": "border-box",
      "display": "flex",
      "height": 56,
      "justifyContent": "space-between",
      "muiPrepared": true,
      "padding": "0px 24px",
      "paddingRight": 0,
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "auto",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 48,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 12,
          "color": "rgba(0, 0, 0, 0.3)",
          "muiPrepared": true,
          "opacity": 0,
          "position": "absolute",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      Search...
    </div>
    <input
      disabled={false}
      id="1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "WebkitOpacity": 1,
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="text"
      value="q1"
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 1,
            "borderColor": "#e0e0e0",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 2,
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;

exports[`StackHeader renders canRefresh & loading 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "rgb(232, 232, 232)",
      "boxSizing": "border-box",
      "display": "flex",
      "height": 56,
      "justifyContent": "space-between",
      "muiPrepared": true,
      "padding": "0px 24px",
      "paddingRight": 0,
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "not-allowed",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 48,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 12,
          "color": "rgba(0, 0, 0, 0.3)",
          "muiPrepared": true,
          "opacity": 0,
          "position": "absolute",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      Search...
    </div>
    <input
      disabled={true}
      id="1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "WebkitOpacity": 1,
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "color": "rgba(0, 0, 0, 0.3)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="text"
      value="q1"
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "dotted",
            "borderBottomWidth": 2,
            "borderColor": "rgba(0, 0, 0, 0.3)",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 2,
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
          />
        </svg>
      </div>
    </button>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;

exports[`StackHeader renders canRefresh 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "rgb(232, 232, 232)",
      "boxSizing": "border-box",
      "display": "flex",
      "height": 56,
      "justifyContent": "space-between",
      "muiPrepared": true,
      "padding": "0px 24px",
      "paddingRight": 0,
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "auto",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 48,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 12,
          "color": "rgba(0, 0, 0, 0.3)",
          "muiPrepared": true,
          "opacity": 0,
          "position": "absolute",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      Search...
    </div>
    <input
      disabled={false}
      id="1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "WebkitOpacity": 1,
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="text"
      value="q1"
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 1,
            "borderColor": "#e0e0e0",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 2,
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
          />
        </svg>
      </div>
    </button>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;

exports[`StackHeader renders loading 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "rgb(232, 232, 232)",
      "boxSizing": "border-box",
      "display": "flex",
      "height": 56,
      "justifyContent": "space-between",
      "muiPrepared": true,
      "padding": "0px 24px",
      "paddingRight": 0,
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "not-allowed",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 48,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <div
      style={
        Object {
          "bottom": 12,
          "color": "rgba(0, 0, 0, 0.3)",
          "muiPrepared": true,
          "opacity": 0,
          "position": "absolute",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      Search...
    </div>
    <input
      disabled={true}
      id="1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "WebkitOpacity": 1,
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "color": "rgba(0, 0, 0, 0.3)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="text"
      value="q1"
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "dotted",
            "borderBottomWidth": 2,
            "borderColor": "rgba(0, 0, 0, 0.3)",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 2,
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;
